<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1032</width>
    <height>685</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Qt5 Image Viewer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QFormLayout" name="formLayout">
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QPushButton" name="btnLoadBG">
            <property name="text">
             <string>Load Image</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="grayscale">
            <property name="text">
             <string>RGBtoGRAYSCALE (CPU)</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="globalBinarize">
            <property name="text">
             <string>GlobalBinarize (CPU)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="globalThresholdSlider">
            <property name="maximum">
             <number>256</number>
            </property>
            <property name="sliderPosition">
             <number>128</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="GlobalThresholdText">
            <property name="text">
             <string>GLOBAL threshold value: 128</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="NickCPU">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>NICK method (local binaritzation) (CPU)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="NickGPU">
            <property name="styleSheet">
             <string notr="true">background-color: rgb(0, 255, 0)</string>
            </property>
            <property name="text">
             <string>NICK method (local binaritzation) (GPU)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="windowSizeSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
            <property name="pageStep">
             <number>2</number>
            </property>
            <property name="value">
             <number>6</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="windowSizeText">
            <property name="text">
             <string>width size of window: 6</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="evenNumber">
            <property name="text">
             <string>window size has to be an EVEN number!</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="kSlider">
            <property name="minimum">
             <number>-20</number>
            </property>
            <property name="maximum">
             <number>-10</number>
            </property>
            <property name="value">
             <number>-20</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="kText">
            <property name="text">
             <string>k value: -0.2</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Scrolling(V) -&gt; Mouse Wheel
Scrolling(H) -&gt; Shift+Wheel
Zoom -&gt; Ctrl+Wheel</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnFitWindow">
            <property name="text">
             <string>Fit to window</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QTextBrowser" name="textBrowser">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>0</width>
          <height>100</height>
         </size>
        </property>
        <property name="acceptDrops">
         <bool>true</bool>
        </property>
        <property name="acceptRichText">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="MyGraphicsView" name="m_graphicsView">
        <property name="minimumSize">
         <size>
          <width>800</width>
          <height>600</height>
         </size>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>mygraphicsview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>btnLoadBG</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onPressLoadImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnFitWindow</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onPressFitWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>52</x>
     <y>498</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>windowSizeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onWindowSizeSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>276</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>kSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onkSilderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grayscale</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onGrayscaleClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>globalThresholdSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onGlobalThresholdSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>globalBinarize</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onGlobalBinarizeClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NickGPU</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onNickGPUClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NickCPU</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onNickCPUClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>224</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onPressLoadImage()</slot>
  <slot>onPressFitWindow()</slot>
  <slot>onPressSaveImage()</slot>
  <slot>onWindowSizeSliderChanged(int)</slot>
  <slot>onkSilderChanged(int)</slot>
  <slot>onGrayscaleClicked()</slot>
  <slot>onGlobalThresholdSliderChanged(int)</slot>
  <slot>onGlobalBinarizeClicked()</slot>
  <slot>onNickGPUClicked()</slot>
  <slot>onNickCPUClicked()</slot>
 </slots>
</ui>
